src/map/mob.cpp

```diff
@@ -1154,7 +1154,10 @@ int mob_spawn (struct mob_data *md)
 
 	md->state.aggressive = status_has_mode(&md->status,MD_ANGRY)?1:0;
 	md->state.skillstate = MSS_IDLE;
-	md->next_walktime = tick+rnd()%1000+MIN_RANDOMWALKTIME;
+	if (battle_config.mob_walk_onspawn)
+		md->next_walktime = tick+rnd()%1000+MIN_RANDOMWALKTIME;
+	else
+		md->next_walktime = -1;
```

Comment:
From how I understand it anything < current tick will trigger walking on the next iteration.
So you might want to set it to INT64_MAX instead, since you want to prevent that it triggers walking?
Maybe I have read something wrong though I just looked over it quickly.